### DISCUSSION TITLE
Excessive local storage use

### DISCUSSION THREAD

**Discussion by samip5** [k3s-io/k3s#6486] [Q&A]
**Environmental Info:**
K3s Version: v1.24.4+k3s1

Node(s) CPU architecture, OS, and Version: 
arm64, Ubuntu 22.04
amd64, Ubuntu 22.04

Cluster Configuration:
1 server, 5 agents

**Describe the bug:**
When the node only has ~32GB of eMMC storage, it's very quickly eaten by /var/lib/rancher/k3s/agent/containerd.

**Steps To Reproduce:**

- Installed K3s: Using the ansible role at https://github.com/PyratLabs/ansible-role-k3s

**Expected behavior:**
I would expect that it wouldn't leave ~10-15GB of things laying around.

**Actual behavior:**
It leaves ~10GB of things laying around, and it usually doesn't seem to be able to fee storage.

**Additional context / logs:**
```
root@k8s-worker7:~# du -sh /var/lib/rancher/k3s/agent/containerd/* |sort -h
4.0K    /var/lib/rancher/k3s/agent/containerd/bin
4.0K    /var/lib/rancher/k3s/agent/containerd/io.containerd.runtime.v1.linux
4.0K    /var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.btrfs
4.0K    /var/lib/rancher/k3s/agent/containerd/lib
4.0K    /var/lib/rancher/k3s/agent/containerd/tmpmounts
8.0K    /var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.fuse-overlayfs
8.0K    /var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.native
32K     /var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.stargz
384K    /var/lib/rancher/k3s/agent/containerd/io.containerd.runtime.v2.task
6.6M    /var/lib/rancher/k3s/agent/containerd/containerd.log
7.2M    /var/lib/rancher/k3s/agent/containerd/io.containerd.metadata.v1.bolt
409M    /var/lib/rancher/k3s/agent/containerd/io.containerd.grpc.v1.cri
3.0G    /var/lib/rancher/k3s/agent/containerd/io.containerd.content.v1.content
9.4G    /var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs
```

**Comment by brandond** [comment]
By default the kubelet aggressively caches images until the disk holding the image store is about 85% full  You'd want to tune the `--image-gc-high-threshold` and `--image-gc-low-threshold` values if this does not work for you. See the kubelet documentation for more details:
https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/

You can use crictl to access containerd to see what images are stored locally on the node.
